### PR TITLE
[WFLY-13226]: Upgrade Apache Artemis to 2.16.0

### PR DESCRIPTION
**Pull request by ehsavoie**
* Upgrading Apache Artemis version to 2.16.0
 * journal now needs to be able to load json.
 * Adding support for the SSLProviderFactory
 * Fixing change in BroadcastGroupControlHandler due to ARTEMIS-2749.

Jira: https://issues.redhat.com/browse/WFLY-13226

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalConnectionFactoryAdd.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final String name = context.getCurrentAddressValue();
         final ServiceName serviceName = ExternalConnectionFactoryDefinition.CAPABILITY.getCapabilityServiceName(context.getCurrentAddress());
         boolean ha = HA.resolveModelAttribute(context, model).asBoolean();
-        boolean enable1Prefixes = ENABLE_AMQ1_PREFIX.resolveModelAttribute(context, model).asBoolean();
+        boolean enable1Prefixes = ENABLE_AMQ1_PREFIX.resolveModelAttribute(context, model).asBoolean(true);
```
Comment (bstansberry) [inline messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalConnectionFactoryAdd.java:79]: This is not needed as the AttributeDefinition has default value ModelNode.TRUE.

File: pom.xml [hunk 2]
```diff
@@ -1,25 +1,25 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-  ~ JBoss, Home of Professional Open Source.
```
Comment (bstansberry) [inline pom.xml:3]: Do you want all the whitespace change here?

I guess it's ok but maybe in a separate commit?

Comment (ehsavoie) [reply pom.xml:3]: no i'll remove those

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalConnectionFactoryAdd.java (hunk 1 above)
Comment (ehsavoie) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalConnectionFactoryAdd.java:79]: sorry, leftover

**Comment by ehsavoie** [comment]
@bstansberry PR fixed

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
@ehsavoie It seems the merge of the network health RFE broke this. Or this breaks the test for network health.

**Comment by ehsavoie** [comment]
@bstansberry they have fixed a trypo in the log message we were checking ;)
It should be passing.

**Comment by FireBurn** [comment]
I'm struggling to compile wildfly 22 with Java 8 - it appears artemis 2.16.0 is compiled with java 11
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /apps/was/jenkins/workspace/Compile_WildFly/wildfly-22.0.0.Final/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java:[37,51] cannot access org.apache.activemq.artemis.api.core.client.ActiveMQClient
  bad class file: /apps/was/.m2/repository/org/apache/activemq/artemis-core-client/2.16.0/artemis-core-client-2.16.0.jar(org/apache/activemq/artemis/api/core/client/ActiveMQClient.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
[ERROR] /apps/was/jenkins/workspace/Compile_WildFly/wildfly-22.0.0.Final/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java:[35,1] cannot find symbol
  symbol:   static noDuplicateElements
  location: class
[INFO] 2 errors

**Pull request merged**